server/src-lib/Hasura/RQL/Types/SchemaCache.hs

```diff
@@ -153,11 +154,17 @@ mkColDep :: DependencyReason -> QualifiedTable -> PGCol -> SchemaDependency
 mkColDep reason tn col =
   flip SchemaDependency reason . SOTableObj tn $ TOCol col
 
+mkComputedColumnDep
+  :: DependencyReason -> QualifiedTable -> ComputedColumnName -> SchemaDependency
+mkComputedColumnDep reason tn computedColumn =
+  flip SchemaDependency reason . SOTableObj tn $ TOComputedColumn computedColumn
+
 type WithDeps a = (a, [SchemaDependency])
 
 data FieldInfo columnInfo
   = FIColumn !columnInfo
   | FIRelationship !RelInfo
+  | FIComputedColumn !ComputedColumnInfo
```

Comment:
I’m curious: do you think it would make any sense to slot computed columns into `PGColumnType` directly, as another case in addition to `PGColumnScalar` and `PGColumnEnumReference`? I can think of arguments both for and against: on one hand, computed columns are genuinely virtual *columns*, while relationships are an entirely different sort of thing, so it might make sense to model them that way, but on the other hand, they’re a lot less like a plain old column than enum references are. I’d be curious to hear your thoughts.

Reply:
I'm going to rename `computed column` to `computed field` to make it distinct from Postgres column. In that way, maybe, I think `FIComputedField` (new name) is good enough to be isolated.

Reply:
I think that renaming is a good idea, though I’m still curious about whether you think it makes sense for them to be grouped in with columns (as, really, I think I’m a little inclined to think that maybe `PGColumnType` ought to have been called `FieldType`, instead, and we should reserve “column” for the Postgres concept 🙂).

Reply:
I thought in this way. The `FieldInfo` should show the kinds of fields that a table can have. Columns (from Postgres), relationships, computed fields (this PR), remote join fields (coming soon) etc. Except for columns all other fields are virtual and only defined through Hasura metadata (though enums are defined through metadata but the columns are actually present in table schema). So, I'm thinking it's better to keep `FIComputedField` isolated and do not group with `FIColumn` :)

Reply:
Okay, I think that’s fair, you’ve convinced me. :) Reading your comment made me realize the distinction here: `PGColumnType` describes the type of a value, while `FieldInfo` describes where we get it from. And the value of a computed column is not a different type of value from the value of an ordinary column! The way we retrieve it is just different.